src/sage/categories/finite_monoids.py

```diff
@@ -311,3 +315,21 @@ def __invert__(self):
                 return next(it)
             except StopIteration:
                 raise ValueError(f"the element {self} is not invertible")
+
+    class CartesianProducts(CartesianProductsCategory):
+        class ElementMethods:
+            # needed otherwise __invert__ from FiniteMonoids will override this
+            __invert__ = sage.categories.magmas.Magmas.Unital.CartesianProducts.ElementMethods.__invert__
+
+        def extra_super_categories(self):
+            r"""
+            Implement the fact that this structure is stable under (finite) Cartesian
+            products. Also to ensure this base class appears before ``FiniteMonoids``.
+
+            TESTS::
+
+                sage: from sage.categories.rngs import Rngs
+                sage: FiniteMonoids().CartesianProducts().extra_super_categories()
+                [Category of finite monoids]
+            """
+            return [FiniteMonoids()]
```

Comment:
note that currently only finite Cartesian products are implemented, so a product of finitely many finite monoids is a finite monoid.